apps/rule-manager/app/controllers/RulesController.scala

```diff
@@ -279,4 +279,29 @@ class RulesController(
       case Left(error) => BadRequest(s"Invalid request: $error")
     }
   }
+
+  def csvImport() = Action { implicit request =>
+    val rules = for {
+      formData <- request.body.asMultipartFormData.toRight("No form data found in request")
+      file <- formData.file("file").toRight("No file found in request")
+      tag = formData.dataParts.get("tag") match {
+        case Some(tag) => Some(tag.head)
+        case None      => None
+      }
+      category = formData.dataParts.get("category") match {
+        case Some(category) => Some(category.head)
+        case None           => None
+      }
```

Comment:
I think this can be even neater:

```suggestion
      formData <- request.body.asMultipartFormData.toRight("No form data found in request")
      file <- formData.file("file").toRight("No file found in request")
      tag = formData.dataParts.get("tag").flatMap(_.headOption)
      category = formData.dataParts.get("category").flatMap(_.headOption)
```

Reply:
Lovely - thank you!